File: backend/src/main/java/com/sliit/backend/notificationService.java
package com.sliit.backend;
 
import java.util.List;
 
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.data.mongodb.core.query.Criteria;
import org.springframework.data.mongodb.core.query.Query;
import org.springframework.data.mongodb.core.query.Update;
import org.springframework.stereotype.Service;
 
@Service
public class notificationService {
 
    @Autowired
    private notificationRepository notificationRepository;
 
    @Autowired
    private MongoTemplate mongoTemplate;
 
    // Fetch all notifications for a user (Receiver)
    public List<Notification> getNotificationsForUser(String userId) {
        return notificationRepository.findByReceiverUserIdOrderByTimestampDesc(userId);
    }
 
    // Delete a notification by its ID
    public void deleteNotification(String id) {
        notificationRepository.deleteById(id);
    }
 
    // Mark a single notification as read
    public void markNotificationAsRead(String id) {
        Query query = new Query(Criteria.where("id").is(id));
        Update update = new Update().set("isRead", true);
        mongoTemplate.updateFirst(query, update, Notification.class);
    }
 
    // Mark all notifications as read for a user
    public void markAllNotificationsAsRead(String userId) {
        Query query = new Query(Criteria.where("receiverUserId").is(userId));
        Update update = new Update().set("isRead", true);
        mongoTemplate.updateMulti(query, update, Notification.class);
    }
 
    // Create a new notification
    public Notification createNotification(String message, String postId, String receiverUserId, String senderUserId) {
        Notification notification = new Notification(message, postId, receiverUserId, senderUserId);
        return notificationRepository.save(notification);
    }
}

File: frontend/src/services/NotificationService.js
// src/services/NotificationService.js
import axios from 'axios';

const API_URL = 'http://localhost:5021/api/notifications/';

class NotificationService {
    static async getNotifications(userId) {
        try {
            const response = await axios.get(API_URL + userId);
            return response.data;
        } catch (error) {
            console.error('Error fetching notifications', error);
            throw error;
        }
    }

    static async markNotificationAsRead(notificationId) {
        try {
            await axios.put(API_URL + notificationId + '/read');
        } catch (error) {
            console.error('Error marking notification as read', error);
            throw error;
        }
    }

    static async markAllNotificationsAsRead(userId) {
        try {
            await axios.put(API_URL + userId + '/read-all');
        } catch (error) {
            console.error('Error marking all notifications as read', error);
            throw error;
        }
    }
}

export default NotificationService;